File: sample/python/external_lib_helper.py
# external_lib_helper.py - 외부 라이브러리 사용 테스트

import numpy as np
from datetime import datetime, timedelta


def process_array(arr):
    """numpy 배열 처리"""
    return np.array(arr) * 2


def calculate_mean(arr):
    """numpy mean 사용"""
    return np.mean(arr)


def calculate_sum(arr):
    """numpy sum 사용"""
    return np.sum(arr)


def add_days(date, days):
    """timedelta 사용"""
    return date + timedelta(days=days)


def format_date(date):
    """날짜 포맷팅"""
    return date.strftime('%Y-%m-%d')


def process_data(data):
    """여러 numpy 함수를 함께 사용"""
    arr = np.array(data)
    mean_val = np.mean(arr)
    sum_val = np.sum(arr)
    return {'mean': mean_val, 'sum': sum_val}

File: sample/nodejs/externalLibHelper.js
// externalLibHelper.js - 외부 라이브러리 사용 테스트

const _ = require('lodash');
const moment = require('moment');

/**
 * lodash의 _.map 사용
 */
function processArray(arr) {
    return _.map(arr, item => item * 2);
}

/**
 * lodash의 _.filter 사용
 */
function filterEvenNumbers(arr) {
    return _.filter(arr, num => num % 2 === 0);
}

/**
 * lodash의 _.sum 사용
 */
function calculateSum(arr) {
    return _.sum(arr);
}

/**
 * moment.js 사용 - 날짜 포맷팅
 */
function formatDate(date) {
    return moment(date).format('YYYY-MM-DD');
}

/**
 * moment.js 사용 - 날짜 계산
 */
function addDays(date, days) {
    return moment(date).add(days, 'days').toDate();
}

/**
 * 여러 lodash 함수를 함께 사용
 */
function processData(data) {
    const filtered = _.filter(data, item => item.active);
    const mapped = _.map(filtered, item => item.value);
    return _.sum(mapped);
}

module.exports = {
    processArray,
    filterEvenNumbers,
    calculateSum,
    formatDate,
    addDays,
    processData
};
